MERLin/MERLin/Routing/RouterProtocol.swift

```diff
@@ -236,14 +277,14 @@ public extension Router {
                 }
             #endif
             
-            if shouldPush,
-                let navigationController = currentController as? UINavigationController ??
-                (currentController as? UITabBarController)?.selectedViewController as? UINavigationController {
+            switch behaviour.presentationStyle {
+            case .pushIfPossible:
+                guard let navigationController = currentController as? UINavigationController ??
+                    (currentController as? UITabBarController)?.selectedViewController as? UINavigationController else { fallthrough }
                 navigationController.pushViewController(deeplinkedViewController, animated: animated)
-                
                 os_log("🔗 Pushed view controller %@ for deeplink %@", log: .router, type: .debug, deeplinkedViewController, deeplink)
-            } else {
-                if shouldPush {
+            case .alwaysModally:
+                if behaviour.presentationStyle == .pushIfPossible {
```

Comment:
how is it possible inside case `.alwaysModally` `.pushIfPossible` should never happen

Reply:
It can happen. If the preference is push**IfPossible** and the currentController is not, or does not contain a navigation controller, then the only way is modal presentation. It all depends on the current context of the current View stack